### ISSUE TITLE
Support changing logging level of auth service launched by Cattle

### ISSUE THREAD

**Issue by prachidamle** [rancher/rancher#9390]
**Rancher versions:**
rancher/server: v1.6

During rancher/server startup, Cattle launches few microservices like rancher-auth-service , catalog-service etc.

The logging level of rancher-auth-service cannot be changed at runtime and often hampers debugging customer issues. 

We should be able to change the logging level of the auth-service on the fly.

**Comment by sabiodelhielo** [comment]
Verified change levels produces the expected level output to the logs:

Updated to DEBUG:
HTTP Request:
HTTP/1.1 PUT /v2-beta/settings/auth.service.log.level
Host: 165.227.11.10:8080
Accept: application/json
Content-Type: application/json
Content-Length: 120

{
"activeValue": "info",
"id": "auth.service.log.level",
"name": "auth.service.log.level",
"source": "Database",
"value": "debug"
}
HTTP Response:
HTTP/1.1 200
Date: Thu, 20 Jul 2017 23:32:14 GMT
X-Api-User-Id: 1a1
Server: Jetty(9.2.11.v20150529)
X-Api-Client-Ip: 68.106.109.133
Content-Type: application/json; charset=utf-8
X-Api-Account-Id: 1a1
X-Api-Schemas: http://165.227.11.10:8080/v2-beta/schemas
X-Rancher-Version: v1.6.6-rc1
Content-Length: 306
Expires: Thu, 01 Jan 1970 00:00:00 GMT

{
"id": "auth.service.log.level",
"type": "activeSetting",
"links": {
"self": "…/v2-beta/settings/auth.service.log.level"
},
"actions": { },
"baseType": "setting",
"name": "auth.service.log.level",
"activeValue": "info",
"inDb": true,
"source": "Database",
"value": "debug"
}


time="2017-07-20T22:57:18Z" level=debug msg="Update setting key:api.auth.external.provider.configured value: true"
time="2017-07-20T22:57:18Z" level=debug msg="Reload called"
time="2017-07-20T22:57:18Z" level=debug msg="Updated config, listing the config back"
time="2017-07-20T22:57:18Z" level=debug msg="Reload called"
time="2017-07-20T22:57:18Z" level=info msg="Reload config is already in process, skipping"
time="2017-07-20T22:57:18Z" level=debug msg="Settings from db for provider github: map[github:map[api.auth.github.client.id:4bfe3bd54bf6de400ac0 api.auth.github.client.secret:a75a66d8bf0526c59caadae859c7a279c6121c9f api.github.domain: api.github.scheme:https://]]"
time="2017-07-20T22:57:18Z" level=debug msg="Provider settings: map[string]string{\"api.auth.github.client.id\":\"4bfe3bd54bf6de400ac0\", \"api.auth.github.client.secret\":\"a75a66d8bf0526c59caadae859c7a279c6121c9f\", \"api.github.domain\":\"\", \"api.github.scheme\":\"https://\"}"
time="2017-07-20T22:57:18Z" level=info msg=" Auth provider configured githubconfig"
time="2017-07-20T22:57:18Z" level=debug msg="Reload called"
time="2017-07-20T22:57:18Z" level=debug msg="Settings from db for provider github: map[github:map[api.auth.github.client.id:4bfe3bd54bf6de400ac0 api.auth.github.client.secret:a75a66d8bf0526c59caadae859c7a279c6121c9f api.github.domain: api.github.scheme:https://]]"
time="2017-07-20T22:57:18Z" level=debug msg="Provider settings: map[string]string{\"api.auth.github.client.id\":\"4bfe3bd54bf6de400ac0\", \"api.auth.github.client.secret\":\"a75a66d8bf0526c59caadae859c7a279c6121c9f\", \"api.github.domain\":\"\", \"api.github.scheme\":\"https://\"}"
time="2017-07-20T22:57:18Z" level=debug msg="Settings from db for provider github: map[github:map[api.auth.github.client.id:4bfe3bd54bf6de400ac0 api.auth.github.client.secret:a75a66d8bf0526c59caadae859c7a279c6121c9f api.github.domain: api.github.scheme:https://]]"
time="2017-07-20T22:57:18Z" level=debug msg="Provider settings: map[string]string{\"api.auth.github.client.id\":\"4bfe3bd54bf6de400ac0\", \"api.auth.github.client.secret\":\"a75a66d8bf0526c59caadae859c7a279c6121c9f\", \"api.github.domain\":\"\", \"api.github.scheme\":\"https://\"}"
time="2017-07-20T22:57:18Z" level=info msg=" Auth provider configured githubconfig"
time="2017-07-20T22:57:31Z" level=debug msg="Reload called"
time="2017-07-20T22:57:31Z" level=debug msg="Settings from db for provider github: map[github:map[api.auth.github.client.id:4bfe3bd54bf6de400ac0 api.auth.github.client.secret:a75a66d8bf0526c59caadae859c7a279c6121c9f api.github.domain: api.github.scheme:https://]]"
time="2017-07-20T22:57:31Z" level=debug msg="Provider settings: map[string]string{\"api.auth.github.client.id\":\"4bfe3bd54bf6de400ac0\", \"api.auth.github.client.secret\":\"a75a66d8bf0526c59caadae859c7a279c6121c9f\", \"api.github.domain\":\"\", \"api.github.scheme\":\"https://\"}"
time="2017-07-20T22:57:31Z" level=info msg=" Auth provider configured githubconfig"
time="2017-07-20T23:00:29Z" level=debug msg="Reload called"
time="2017-07-20T23:00:29Z" level=debug msg="Settings from db for provider github: map[github:map[api.github.scheme:https:// api.auth.github.client.id:4bfe3bd54bf6de400ac0 api.auth.github.client.secret:a75a66d8bf0526c59caadae859c7a279c6121c9f api.github.domain:]]"
time="2017-07-20T23:00:29Z" level=debug msg="Provider settings: map[string]string{\"api.auth.github.client.id\":\"4bfe3bd54bf6de400ac0\", \"api.auth.github.client.secret\":\"a75a66d8bf0526c59caadae859c7a279c6121c9f\", \"api.github.domain\":\"\", \"api.github.scheme\":\"https://\"}"
time="2017-07-20T23:00:29Z" level=info msg=" Auth provider configured githubconfig"



Updated to fatal:

HTTP/1.1 PUT /v2-beta/settings/auth.service.log.level
Host: 165.227.11.10:8080
Accept: application/json
Content-Type: application/json
Content-Length: 127

{
"activeValue": "DEBUGsdfsdf",
"id": "auth.service.log.level",
"name": "auth.service.log.level",
"source": "Database",
"value": "fatal"
}



HTTP/1.1 200
Date: Thu, 20 Jul 2017 23:08:09 GMT
X-Api-User-Id: 1a1
Server: Jetty(9.2.11.v20150529)
X-Api-Client-Ip: 68.106.109.133
Content-Type: application/json; charset=utf-8
X-Api-Account-Id: 1a1
X-Api-Schemas: http://165.227.11.10:8080/v2-beta/schemas
X-Rancher-Version: v1.6.6-rc1
Content-Length: 313
Expires: Thu, 01 Jan 1970 00:00:00 GMT

{
"id": "auth.service.log.level",
"type": "activeSetting",
"links": {
"self": "…/v2-beta/settings/auth.service.log.level"
},
"actions": { },
"baseType": "setting",
"name": "auth.service.log.level",
"activeValue": "DEBUGsdfsdf",
"inDb": true,
"source": "Database",
"value": "fatal"
}

Verified that none of the regular info or debug level logging appear during 
messages that were info or debug did not show up while editting github


Updated to invalid type, logging level does change from previous level, but the API returns the it as the active level:
{
"id": "auth.service.log.level",
"type": "activeSetting",
"links": {
"self": "…/v2-beta/settings/auth.service.log.level"
},
"actions": { },
"baseType": "setting",
"name": "auth.service.log.level",
"activeValue": "asdasdsad",
"inDb": true,
"source": "Database",
"value": "asdasdsad"
}

**Comment by deniseschannon** [comment]
Available in v1.6.8-rc1

**Comment by sangeethah** [comment]
Tested with v1.6.8-rc4

auth.service.log.level is set to "INFO" by default. With this log setting , we get "Info" and "Error" level logs.

```
time="2017-08-18T18:21:49Z" level=info msg="Now generating Ldap token" 
time="2017-08-18T18:23:12Z" level=info msg="Stack Create Event Received" eventId=cd1d0dec-42d4-4188-a6fb-eefd67993cbd resourceId=1st9 
time="2017-08-18T18:23:12Z" level=info msg="Stack Create Event Done" eventId=cd1d0dec-42d4-4188-a6fb-eefd67993cbd resourceId=1st9 
time="2017-08-18T18:52:11Z" level=info msg="Now generating Ldap token" 
time="2017-08-18T18:52:11Z" level=error msg="GetToken failed with error: Error in ldap bind: LDAP Result Code 49 \"Invalid Credentials\": 80090308: LdapErr: DSID-0C0903D9, comment: AcceptSecurityContext error, data 52e, v2580\x00" 
2017-08-18 18:52:11,118 ERROR [:] [] [] [] [p1020391880-135] [.a.a.i.e.ExternalServiceAuthProvider] Got error from Auth service. statusCode: 401, message: Error in ldap bind: LDAP Result Code 49 "Invalid Credentials": 80090308: LdapErr: DSID-0C0903D9, comment: AcceptSecurityContext error, data 52e, v2580 
time="2017-08-18T18:52:18Z" level=info msg="Now generating Ldap token" 
```

When auth.service.log.level is set to "DEBUG", we get "Debug", "Info" and "Error" level logs.

```
time="2017-08-18T20:17:52Z" level=debug msg="CreateToken called with securityCode" 
time="2017-08-18T20:17:52Z" level=info msg="Now generating Ldap token" 
time="2017-08-18T20:17:52Z" level=debug msg="Now creating Ldap connection" 
time="2017-08-18T20:17:52Z" level=debug msg="Binding username password" 
time="2017-08-18T20:17:52Z" level=debug msg="LDAP Search query: {(sAMAccountName=testuser1)}" 
time="2017-08-18T20:17:52Z" level=debug msg="ADConstants userMemberAttribute() {memberOf}" 
time="2017-08-18T20:17:52Z" level=debug msg="SearchResult memberOf attribute {[CN=testgroup1,CN=Users,DC=testad,DC=rancher,DC=io]}" 
time="2017-08-18T20:17:52Z" level=debug msg="Failed to determine if object is type: ldap_group" 
time="2017-08-18T20:17:52Z" level=debug msg="Failed to determine if object is type: person" 
```
